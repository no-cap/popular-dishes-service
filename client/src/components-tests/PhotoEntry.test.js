import React from 'react';
import { shallow, mount } from 'enzyme';
import PhotoEntry from '../components/PhotoEntry.jsx';


var sampleData =
{
    url: 'https://s3-media0.fl.yelpcdn.com/bphoto/EhwQv7W3nmJqdXhvCW7pJg/o.jpg',
    caption: 'Red Dragon roll was so good!! We devoured it in record timing!'
}



describe('PhotoEntry Unit Tests', () => {

    describe('PhotoEntry should render correctly', () => {
        test('PhotoEntry should render correctly in "debug" mode', () => {

            const wrapper = shallow(<PhotoEntry photo={sampleData} />);

            expect(wrapper).toMatchSnapshot();
        });

        test('PhotoEntry should have correct caption and photo url', () => {

            const wrapper = mount(<PhotoEntry photo={sampleData} />);

            expect(wrapper.find('.photo-caption').text()).toEqual('Red Dragon roll was so good!! We devoured it in record timing!')
            expect(wrapper.find('img').prop('src')).toEqual('https://s3-media0.fl.yelpcdn.com/bphoto/EhwQv7W3nmJqdXhvCW7pJg/o.jpg');
        });
    });

});

