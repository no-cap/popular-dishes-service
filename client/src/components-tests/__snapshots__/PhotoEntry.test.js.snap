// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoEntry Unit Tests PhotoEntry should render correctly PhotoEntry should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
